src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -383,8 +434,26 @@ public static bool TryText(FormattingInfo formatInfo, IRContext irContext, Formu
                     }
                     else
                     {
-                        var normalized = dec.Normalize();
-                        result = new StringValue(irContext, normalized.ToString(formatString ?? "g", culture));
+                        var normalized = dec.Normalize();
+
+                        if (!string.IsNullOrEmpty(formatString))

```

Comment:
>if [](http://example.com/codeflow?start=24&length=2)

Can this share with above?
Both decimal and double implement common 
IFormattable

Reply:
updated